extensions/bmrks/src/create-bookmark.tsx

```diff
@@ -0,0 +1,166 @@
+    }
+  }, [activeTab]);
+
+  const valueIsUrl = isUrlLike(value);
+
+  const activeGroup = groups.find((group) => group.id === activeGroupId);
+
+  async function handleSubmit({ groupId, value }: { groupId: string; value: string; title: string }) {
+    if (!value) {
+      await showToast({ style: Toast.Style.Failure, title: "Missing URL", message: "Please provide one" });
+      return;
+    }
+
+    const isValidColor = Boolean(colorString.get(value));
+
+    if (isValidColor) {
+      const res = await db.insertBookmark({
+        type: "color",
+        group_id: groupId,
+        title: value,
+      });
+
+      if (res.error) {
+        await showToast({ style: Toast.Style.Failure, title: "Something went wrong", message: res.error.message });
+        return;
+      }
+    } else if (valueIsUrl) {
+      const validUrl = ensureValidUrl(value);
+      const favicon = await getFavicon(validUrl);
+
+      let title: string | undefined;
+
+      if (!titleValue) {
+        const response = await fetch(`https://api.microlink.io?url=${encodeURIComponent(validUrl)}&data.title`);
+        if (response.ok) {
+          const data = (await response.json()) as MicrolinkResponse;
+          title = data.data.title;
+        }
+      } else {
+        title = titleValue;
+      }
+
+      const res = await db.insertBookmark({
+        type: "link",
+        group_id: groupId,
+        url: validUrl,
+        title,
+        // @ts-expect-error: looks like source is missing
+        favicon_url: favicon.source || "",
+      });
+
+      if (res.error) {
+        await showToast({ style: Toast.Style.Failure, title: "Something went wrong", message: "res.error.message" });
+        return;
+      }
+    } else {
+      const res = await db.insertBookmark({
+        type: "text",
+        group_id: groupId,
+        title: value,
+      });
+
+      if (res.error) {
+        await showToast({ style: Toast.Style.Failure, title: "Something went wrong", message: "res.error.message" });
+        return;
+      }
+    }
+
+    await showHUD(`Saved to ${activeGroup!.name}`, {
+      clearRootSearch: true,
+      popToRootType: PopToRootType.Immediate,
+    });
+  }
+
+  return (
+    <Form
+      isLoading={isLoadingGroups}
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm title="Save Bookmark" onSubmit={handleSubmit} />
+          {activeGroup && user && (
+            <Action.OpenInBrowser
+              url={`https://bmrks.com/${user.user_metadata["username"]}/${activeGroup.name.toLowerCase()}`}
+            />
+          )}
+        </ActionPanel>
+      }
+    >
+      <Form.Dropdown id="groupId" title="Group" onChange={setActiveGroupId}>
+        {groups.map((group) => (
+          <Form.Dropdown.Item key={group.id} value={group.id} title={group.name} />
+        ))}
+      </Form.Dropdown>
+      <Form.Separator />
+      <Form.TextField id="value" autoFocus title="Link, color, or text" value={value} onChange={setValue} />
```

Comment:
```suggestion
      <Form.TextField id="value" autoFocus title="Input" value={value} placeholder="Link, color, or text" onChange={setValue} />
```
Could maybe be something else for title, not sure I like that one I made 😂

Reply:
Let me try it locally